pkg/apis/ext.cattle.io/v1/types.go

```diff
@@ -26,3 +27,147 @@ type UserActivityStatus struct {
+	// different user is rejected as forbidden.
+	// +optional
+	UserID string `json:"userID,omitempty"`
+	// UserPrincipal holds the information about the ext auth provider
+	// managed user (principal) owning the token.
+	UserPrincipal TokenPrincipal `json:"userPrincipal""`
+	// Kind describes the nature of the token. The value "session" indicates
+	// a login/session token. Any other value, including the empty string,
+	// which is the default, stands for some kind of derived token.
+	// +optional
+	Kind string `json:"kind"`
+	// Human readable free-form description of the token. For example, its purpose.
+	// +optional
+	Description string `json:"description, omitempty"`
+	// TTL is the time-to-live of the token, in milliseconds.
+	// The default, 30 days, is indicated by the value `0`.
+	// +optional
+	TTL int64 `json:"ttl"`
+	// Enabled indicates an active token. The default (`null`) indicates an
+	// enabled token.
+	// +optional
+	Enabled *bool `json:"enabled,omitempty"`
+}
+
+// TokenPrincipal contains the data about the user principal owning the token.
+type TokenPrincipal struct {
+	// Name is the name unique to the authentication provider.
+	Name string `json:"name"`
+	// DisplayName is the human readable name/description of the principal.
+	// +optional
+	DisplayName string `json:"displayName,omitempty"`
+	// LoginName is the account name of the principal in the managing auth provider.
+	LoginName string `json:"loginName,omitempty"`
+	// ProfilePicture is an url to a picture to use when displaying the principal.
+	// +optional
+	ProfilePicture string `json:"profilePicture,omitempty"`
+	// ProfileURL is not used by the system
+	// +optional
+	ProfileURL string `json:"profileURL,omitempty"`
+	// PrincipalType is the kind of principal. Legal values are "user" and "group".
+	PrincipalType string `json:"principalType,omitempty"`
+	// Me is a virtual flag for use with the dashboard.
+	Me bool `json:"me,omitempty"`
+	// MemberOf is a virtual flag for use with the dashboard.
+	MemberOf bool `json:"memberOf,omitempty"`
+	// Provider is the name of the auth provider managing the principal
+	Provider string `json:"provider,omitempty"`
+	// ExtraInfo contains additional information about the principal.
+	ExtraInfo map[string]string `json:"extraInfo,omitempty"`
+}
+
+// TokenStatus defines the most recently observed status of the Token.
+type TokenStatus struct {
+	// Value is the access key. It is shown only on token creation and not saved.
+	Value string `json:"tokenValue,omitempty"`
+	// Hash is the hash of the Value.
+	Hash string `json:"tokenHash,omitempty"`
+	// Current indicates whether the token was used to authenticate the current request.
+	Current bool `json:"current"`
+	// Expired indicates whether the token has exceeded its TTL.
+	Expired bool `json:"expired"`
+	// ExpiresAt is the token's expiration timestamp or an empty string if the token doesn't expire.
+	ExpiresAt string `json:"expiresAt"`
```

Comment:
Since this is a timestamp should we use a `*metav1.Time` like the `LastUsedAt`?

Reply:
This field simply informs the user about the time. It is not (expected to be) used anywhere in Rancher where having it at as proper time type would be useful.

Reply:
I'm not sure I agree with this, the name `LastUpdateTime` is pretty common across K8s resources (usually in conditions), and in each case, it's a `metav1.Time` value?

It feels like we should retain the convention?

Reply:
Is this comment on the correct line ? From the text it seems to refer to the line below ?